### PR TITLE
examples: use GrpcCleanupRule for test examples

### PR DESCRIPTION
**Pull request by dapengzhang0**

### CODE REVIEW

File: examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java [hunk 1]
```diff
@@ -52,26 +54,34 @@
...
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
     features = new ArrayList<Feature>();
-    // use directExecutor for both InProcessServerBuilder and InProcessChannelBuilder can reduce the
+    // Use directExecutor for both InProcessServerBuilder and InProcessChannelBuilder can reduce the
     // usage timeouts and latches in test. But we still add timeout and latches where they would be
     // needed if no directExecutor were used, just for demo purpose.
     server = new RouteGuideServer(
-        InProcessServerBuilder.forName(uniqueServerName).directExecutor(), 0, features);
+        InProcessServerBuilder.forName(serverName).directExecutor(), 0, features);
```
Comment (zpencer) [inline examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java]: I think we need to register the server for cleanups as well here.

File: examples/src/test/java/io/grpc/examples/routeguide/RouteGuideClientTest.java [hunk 2]
```diff
@@ -76,25 +83,23 @@ public int nextInt(int bound) {
...
-    fakeServer = InProcessServerBuilder.forName(uniqueServerName)
-        .fallbackHandlerRegistry(serviceRegistry).directExecutor().build().start();
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
+    // Use a mutable service registry for later registering the service impl for each test case.
+    grpcCleanup.register(InProcessServerBuilder.forName(serverName)
+        .fallbackHandlerRegistry(serviceRegistry).directExecutor().build().start());
     client =
-        new RouteGuideClient(InProcessChannelBuilder.forName(uniqueServerName).directExecutor());
+        new RouteGuideClient(InProcessChannelBuilder.forName(serverName).directExecutor());
```
Comment (zpencer) [inline examples/src/test/java/io/grpc/examples/routeguide/RouteGuideClientTest.java]: Channel should be registered for cleanup.

File: examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java (hunk 1 above)
Comment (dapengzhang0) [reply examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java]: server is of type `RouteGuideServer`, not `io.grpc.Server`, and application has its own life-cycle management. So we don't need to register the server for cleanup. The same reason for the other comment about client.

Comment (zpencer) [reply examples/src/test/java/io/grpc/examples/routeguide/RouteGuideServerTest.java]: Got it. Then this is fine.

**Comment by dapengzhang0** [comment]
Thanks @zpencer for review.

**Pull request merged**